exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -228,16 +237,18 @@ impl Dispatcher {
     ///
     /// Until the block built on top of `fork` is committed (or if `fork` is discarded in
     /// favor of another block proposal), no blocking is performed.
-    pub(crate) fn commit_artifact(
-        fork: &Fork,
-        artifact: ArtifactId,
-        deploy_spec: Vec<u8>,
-    ) -> Result<(), ExecutionError> {
-        // TODO: revise dispatcher integrity checks [ECR-3743]
+    ///
+    /// # Panics
+    ///
+    /// This method assumes that `deploy_artifact` was previously called for the corresponding
+    /// `ArtifactId` and deployment was completed successfully.
+    /// If any error happens within `commit_artifact`, it is considered either a bug in the
+    /// `Supervisor` service or `Dispatcher` itself, and as a result, this method will panic.
+    pub(crate) fn commit_artifact(fork: &Fork, artifact: ArtifactId, deploy_spec: Vec<u8>) {
```

Comment:
I'm slightly hesitant about removing an error from this method. If I understand correctly, this places certain additional requirements on the supervisor (sort of; since panics and errors are processed in the same way by the dispatcher, breaking these requirements will lead to the rollback rather than to unexpected behavior).

Reply:
IMHO, `Supervisor` is kind of entity on which we can surely trust -- if it behaves incorrectly, we are doomed anyway.
Knowing the context of `commit_artifact` -- artifact should be previously *successfully* deployed, and the deploy should happen not only for this node, but for the whole network, and all of those conditions are checked by the `Supervisor` only -- it makes sense to make `Supervisor` responsible for this method not panicking.
Overall, if this method will panic, then `Supervisor` somehow did let it panic, and this is already a big problem.